### DISCUSSION TITLE
Is it an issue need to be resolved? `+= (("a", 1))`

### DISCUSSION THREAD

**Discussion by dongxuwang** [scala/scala3#20066] [General Discussion]
For `+=` with tuple, why I have to add `()` to wrap the tuple value

```scala
import collection.mutable.ArrayBuffer
val x: ArrayBuffer[(String, Int)] = ArrayBuffer(("a", 1))
// x += ("b", 2) // this wouldn't work
x += (("b", 2))
```

**Comment by EugeneFlesselle** [comment]
Apparently, we untuple tuple arguments of infix operations if the function does not consist only of unary alternatives.

```scala 3
trait Buffer1[A]:
  def += (elem: A): Buffer1[A]

val b1: Buffer1[(String, Int)] = ???
b1 += ("a", 1) // Ok


trait Buffer2[A]:
  def += (elem: A): Buffer2[A]
  def += (elem1: A, elem2: A): Buffer2[A]

val b2: Buffer2[(String, Int)] = ???
b2 += ("b", 2) // Error
```

https://github.com/scala/scala3/blob/6bb6b4382889e3147602020ad860fd187a3be364/compiler/src/dotty/tools/dotc/typer/Applications.scala#L1231-L1239

**Comment by som-snytt** [comment]
Since the alternative `Growable#+=` is deprecated, it would be handy if it were ignored for purposes of infix untupling. Hopefully it can be removed someday.

The current nice syntax is `x += "b" -> 2`.
